rmw_cyclonedds_cpp/CMakeLists.txt

```diff
@@ -72,6 +72,18 @@ ament_target_dependencies(rmw_cyclonedds_cpp
 
 configure_rmw_library(rmw_cyclonedds_cpp)
 
+string(REGEX REPLACE "([0-9]*)\\.([0-9]*)\\.([0-9]*)((|-).*)" "\\1;\\2;\\3;\\5" parsed_rmw_version "${rmw_VERSION}")
+list(GET parsed_rmw_version 0 rmw_version_major)
+list(GET parsed_rmw_version 1 rmw_version_minor)
+list(GET parsed_rmw_version 2 rmw_version_patch)
```

Comment:
No need to parse those manually. CMake already sets `MAJOR` / `MINOR` / `PATCH` variables based on the `rmw_VERSION` variable.

Reply:
Very good call!